homeassistant/components/bthome/__init__.py

```diff
@@ -93,7 +93,12 @@ def process_service_info(
 
     # If payload is encrypted and the bindkey is not verified then we need to reauth
     if data.encryption_scheme != EncryptionScheme.NONE and not data.bindkey_verified:
-        entry.async_start_reauth(hass, data={"device": data})
+        # There is a small risk of race here if its called multiple times
+        # before the task it run since tasks are not eager.
+        hass.async_create_background_task(
+            entry.async_init_reauth(hass, data={"device": data}),
+            f"reauth flow {DOMAIN} {entry.entry_id}",
+        )
```

Comment:
Why do we need to accept the race here but not in the other cases? Is it because this is run from an executor?

Reply:
Its a more of a note to self that we need to refactor later since there is a tiny risk here.